awx/main/models/jobs.py

```diff
@@ -869,15 +871,28 @@ def start_job_fact_cache(self, destination, modification_times, timeout=None):
                 with codecs.open(filepath, 'w', encoding='utf-8') as f:
                     os.chmod(f.name, 0o600)
                     json.dump(host.ansible_facts, f)
+                    written_ct += 1
             except IOError:
                 system_tracking_logger.error('facts for host {} could not be cached'.format(smart_str(host.name)))
                 continue
             # make note of the time we wrote the file so we can check if it changed later
             modification_times[filepath] = os.path.getmtime(filepath)
+        if written_ct:
+            dt = time.time() - start_time
```

Comment:
this is interesting by itself. would be nice to elevate more logging from debugging to info based on performance criteria. If we do this kind of thing often we could build out a general method/system for it

Reply:
I started on a different kind of standard pattern for that here:

https://github.com/ansible/awx/blob/ff6fb3229720a6ba93f18ba287fbe99359a17a26/awx/main/utils/common.py#L1193

but that's insufficient for what I'm trying to do here, because I would really like to include the counts along with the time taken.

Maybe I could expand that method to take an optional log message, which would be the format statement. Then it would have to get the values from the return value... which here, would probably need to be a dictionary. This shouldn't be a problem because this never returned anything before.

Reply:
I developed this further, and put it into the decorator now.